tests/core/data_structures/test_result.py

```
@@ -163,6 +165,323 @@ def test_result_is_pickleable() -> None:
+def test_result_supports_ordered_nested_segment_results() -> None:
+    """An outer program result reuses Result for ordered segment outputs."""
+    analog_params = AnalogSimParams(elapsed_time=0.1, dt=0.1, get_state=True)
+    digital_params = DigitalSimParams(get_state=True)
+    outer = Result(
+        segment_results=[
+            Result(sim_params=analog_params),
+            Result(sim_params=digital_params),
+        ]
+    )
+
+    restored = pickle.loads(pickle.dumps(outer))  # ruff:ignore[suspicious-pickle-usage]  # controlled round-trip
```

Ruff (0.16.1)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)